### PR TITLE
Fix pipeline stage names

### PR DESCRIPTION
**Pull request by MichaelSimons**
The check-base-image-updates pipeline stage name that was added in https://github.com/dotnet/docker-tools/pull/1214 is invalid as Names cannot contain spaces.  Fixing that as well as adding a more descriptive name for the upload-file pipeline stage.

### CODE REVIEW

File: eng/pipelines/check-base-image-updates.yml [hunk 1]
```diff
@@ -27,7 +27,8 @@ extends:
     customBuildTags:
     - ES365AIMigrationTooling
     stages:
-    - stage: Check Base Images
+    - stage:
```
Comment (lbussell) [inline eng/pipelines/check-base-image-updates.yml:30]:
```suggestion
    - stage: CheckBaseImages
```

I haven't seen any documentation that suggests an empty stage or job name will be accepted.

File: eng/pipelines/upload-file.yml [hunk 2]
```diff
@@ -30,7 +30,8 @@ extends:
     customBuildTags:
     - ES365AIMigrationTooling
     stages:
-    - stage: stage
+    - stage:
```
Comment (lbussell) [inline eng/pipelines/upload-file.yml:33]:
```suggestion
    - stage: UploadFile
```

**Review by lbussell — CHANGES_REQUESTED** [review]

File: eng/pipelines/check-base-image-updates.yml (hunk 1 above)
Comment (MichaelSimons) [reply eng/pipelines/check-base-image-updates.yml:30]: FWIW the pipeline works without one.

Comment (lbussell) [reply eng/pipelines/check-base-image-updates.yml]: You learn something new every day.

**Review by mthalman — APPROVED** [review]

**Pull request merged**